src/Microsoft.DotNet.Build.Tasks/ExecWithRetries.cs

```diff
@@ -0,0 +1,124 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Build.Framework;
+using System;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.DotNet.Build.Tasks
+{
+    /// <summary>
+    /// Run a command and retry if the exit code is not 0.
+    /// </summary>
+    public class ExecWithRetries : Microsoft.Build.Utilities.Task, ICancelableTask
```

Comment:
First I tried overriding Exec's `ExecuteTool`, but there's some temp file cleanup that I couldn't find an extension point to tell it not to do. I also couldn't find a way to create the temp files again before the next run.

Then I tried overriding Execute, but running base.Execute multiple times doesn't work because the exit code it tracks in a private variable is never reset to `0`.